daos/restaurant.js

```diff
@@ -0,0 +1,94 @@
+const mongoose = require("mongoose");
+// optional shortcut to the mongoose.Schema class
+const Schema = mongoose.Schema;
+
+// import related daos (to facilitate retrieval of idUser in schemaUser and idBooking in schemaBooking)
+const daoUser = require("./user");
+const daoBooking = require("./booking");
+
+/*{
+    "idRest": "658ac33fcfe93c8dbf44gb99",
+    "imageRest":
+    "name":"testsample",
+    "category": “Japanese”,
+    "location": “East”,
+    "timeOpen": "1100",
+    "timeClose": "2000",
+    "daysClose":"Thursday",
+    "address":"59 Devon Road Singapore 777888",
+    "phone":"88997788",
+    "websiteUrl":"www.testestes.com",
+    "maxPax":"30",
+    "idUser":"658ac33fcfe93c8dbf43fb28"
+    "restDest":"mid-tier jap restaurant in midtown",
+    } */
+
+const schemaRestaurant = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    image: {
+      // need this to be URL to facilitate FE rendering
+      type: String,
+      required: false,
+    },
+    category: {
+      type: String,
+      enum: ["Asian", "Chinese", "Japanese", "Western"],
+      required: true,
+    },
+    location: {
+      type: String,
+      enum: ["North", "South", "East", "West", "Central"],
+      required: true,
+    },
+    timeOpen: {
+      type: Number,
+      /* to consider using string to facilitate input validation. also no need to use Data type as we are not performing any
+        date-related operations (eg, calculating opening duration) */
+    },
+    timeClose: {
+      type: Number,
+    },
+    daysClose: {
+      type: [Date],
+    },
+    address: {
+      type: String,
+    },
+    phone: {
+      type: String,
```

Comment:
Changed phone type from `Number` to `String` for now since the value such as `0123 4567` could not be handled if it is Number type. 
To be discussed further.